### DISCUSSION TITLE
How to write an "if exists, get, otherwise insert"

### DISCUSSION THREAD

**Discussion by 2A5F** [greg7mdp/parallel-hashmap#257] [Q&A]
How to write an "if exists, get, otherwise insert"

**Comment by greg7mdp** [comment]
Which hash_map are you using? And is it in a multi-thread context?

**Reply by 2A5F** [reply]
parallel_flat_hash_map, yes, in a multi-thread context

**Comment by greg7mdp** [comment]
You can use `lazy_emplace_l`.
